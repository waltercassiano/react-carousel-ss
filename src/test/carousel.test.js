import React from "react";
import Carousel from "../lib/carousel.js";
import { mount, render, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { times } from "lodash";

configure({ adapter: new Adapter() });

test("Render Component - default props", () => {
  const component = render(<Carousel />);
  expect(component).toMatchSnapshot();
});

test("Render Component - prev and next active", () => {
  const component = render(<Carousel prev={true} next={true} />);
  expect(component).toMatchSnapshot();
});

test("Render Component - prev and click on prev", () => {
  const component = mount(
    <Carousel prev={true} next={true}>
      <div>1</div>
      <div>2</div>
      <div>3</div>
    </Carousel>
  );

  const currentIndex = component.find("Carousel").getElements()[0].props
    .activeSlide;

  component.find(".rcss-prev").simulate("click");

  const nextIndex = component.find("Carousel").getElements()[0].props
    .activeSlide;

  expect(nextIndex).toBe(currentIndex - nextIndex);
});

test("Render Component - next and click on next", () => {
  const component = mount(
    <Carousel prev={true} next={true}>
      <div>1</div>
      <div>2</div>
      <div>3</div>
    </Carousel>
  );
  const currentIndex = component.find("Carousel").getElements()[0].props
    .activeSlide;

  component.find(".rcss-next").simulate("click");

  const nextIndex = component.find("Carousel").getElements()[0].props
    .activeSlide;

  expect(nextIndex).toBe(currentIndex + nextIndex);
});

test("Render Component - go to last slide then return to the first", () => {
  const component = mount(
    <Carousel prev={true} next={true}>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
      <div>slide</div>
    </Carousel>
  );

  // Go to Last Slide
  const numberSlide =
    component.find("Carousel").getElements()[0].props.numberSlide - 1;

  times(numberSlide, () => {
    component.find(".rcss-next").simulate("click");
  });

  const activeSlideNext = component.find("Carousel").getElements()[0].props
    .activeSlide;
  expect(activeSlideNext).toBe(numberSlide);

  // Return to first Slide
  times(numberSlide, () => {
    component.find(".rcss-prev").simulate("click");
  });

  const activeSlidePrev = component.find("Carousel").getElements()[0].props
    .activeSlide;

  expect(activeSlidePrev).toBe(0);
});

const visibleItems = 1;
test(
  "Render Component - default visible items equal to " + visibleItems,
  () => {
    const component = mount(
      <Carousel prev={true} next={true} showItemsNumber={visibleItems}>
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Carousel>
    );
    const showItemsNumber = component.find("Carousel").getElements()[0].props
      .showItemsNumber;
    expect(visibleItems).toBe(showItemsNumber);
  }
);
