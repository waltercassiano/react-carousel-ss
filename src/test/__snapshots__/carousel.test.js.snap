// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Component - default props 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "position:relative;width:100%;box-sizing:border-box;overflow:hidden;height:100px",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "style": "position:absolute;z-index:100;top:20px;width:100%",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "style": "position:absolute;display:flex;left:0;top:0;box-sizing:border-box;width:0vw",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "style": "position:absolute;display:flex;left:0;top:0;box-sizing:border-box;width:0vw",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "style": "position:absolute;z-index:100;top:20px;width:100%",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Render Component - prev and next active 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "position:relative;width:100%;box-sizing:border-box;overflow:hidden;height:100px",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "style": "position:absolute;z-index:100;top:20px;width:100%",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "rcss-prev",
            },
            "children": Array [
              Object {
                "data": "prev",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "rcss-next",
              },
              "children": Array [
                Object {
                  "data": "prev",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "rcss-next",
            },
            "children": Array [
              Object {
                "data": "prev",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "rcss-prev",
              },
              "children": Array [
                Object {
                  "data": "prev",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "style": "position:absolute;display:flex;left:0;top:0;box-sizing:border-box;width:0vw",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "style": "position:absolute;display:flex;left:0;top:0;box-sizing:border-box;width:0vw",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "style": "position:absolute;z-index:100;top:20px;width:100%",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "rcss-prev",
              },
              "children": Array [
                Object {
                  "data": "prev",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "rcss-next",
                },
                "children": Array [
                  Object {
                    "data": "prev",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "rcss-next",
              },
              "children": Array [
                Object {
                  "data": "prev",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "rcss-prev",
                },
                "children": Array [
                  Object {
                    "data": "prev",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
